utilities/transactions/write_unprepared_txn.cc

```diff
@@ -48,28 +48,35 @@ WriteUnpreparedTxn::WriteUnpreparedTxn(WriteUnpreparedTxnDB* txn_db,
                                        const TransactionOptions& txn_options)
     : WritePreparedTxn(txn_db, write_options, txn_options), wupt_db_(txn_db) {
   max_write_batch_size_ = txn_options.max_write_batch_size;
+  // We set max bytes to zero so that we don't get a memory limit error.
+  // Instead of trying to keep write batch strictly under the size limit, we
```

Comment:
> Instead of trying to keep write batch strictly under the size limit
 I guess we still keep the write batch under the limit we just do it ourselves instead of relying on the WriteBatch class to enforce it.

Reply:
Actually, it's possible for the write batch to exceed the size limit too. For example, if you put a huge key-value, the current algorithm will allow this, and flush the single put, whereas having the limit on will make this impossible.

I'll update the comment, since that was the real reason I went with this approach.